### PR TITLE
Feat/ag 234 random fixes

### PR DESCRIPTION
**Pull request by PlamenHristov**

### CODE REVIEW

File: crates/validation/src/common/token_pricing.rs [hunk 1]
```diff
@@ -13,31 +13,32 @@ use futures::StreamExt;
 use tracing::warn;
 use uniswap_v4::uniswap::{pool_data_loader::PoolDataLoader, pool_manager::SyncedUniswapPools};
 
-const BLOCKS_TO_AVG_PRICE: u64 = 5;
```
Comment (Will-Smith11) [inline crates/validation/src/common/token_pricing.rs:16]: In the Spec we set it to 5 and it shouldn't be configurable. why was this changed?

File: bin/angstrom/src/components.rs [hunk 2]
```diff
@@ -217,18 +218,22 @@ pub async fn initialize_strom_components<Node: FullNodeComponents, AddOns: NodeA
...
-            .expect("failed to start token price generator");
+    let blocks_to_average_price = 5;
+    let price_generator = TokenPriceGenerator::new(
+        provider.clone(),
+        block_id,
+        uniswap_pools.clone(),
+        Some(blocks_to_average_price)
+    )
+    .await
+    .expect("failed to start token price generator");
```
Comment (Will-Smith11) [inline bin/angstrom/src/components.rs:229]: Why did you move this away from the constant. It is a constant in the spec and should always be 5. this implies its configurable which it shouldn't be.

File: testing-tools/src/controllers/devnet/config.rs [hunk 3]
```diff
@@ -77,17 +77,18 @@ impl TestingConfig for DevnetConfig {
...
         let ipc = alloy::providers::IpcConnect::new(endpoint);
         // TODO: Revert after things are working
-        // let sk: PrivateKeySigner = "4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356".parse().unwrap();
-        // let controller_address = "0x14dc79964da2c08b23698b3d3cc7ca32193d9955".parse().unwrap();
-        let sk: PrivateKeySigner = anvil.keys()[self.anvil_key].clone().into();
-        let controller_address = anvil.addresses()[self.anvil_key];
+        let sk: PrivateKeySigner = "4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356".parse().unwrap();
+        let controller_address = "0x14dc79964da2c08b23698b3d3cc7ca32193d9955".parse().unwrap();
+        // let sk: PrivateKeySigner = anvil.keys()[self.anvil_key].clone().into();
+        // let controller_address = anvil.addresses()[self.anvil_key];
```
Comment (Will-Smith11) [inline testing-tools/src/controllers/devnet/config.rs:89]: ? why are we making this less generic?

**Review by Will-Smith11 — CHANGES_REQUESTED** [review]
Q's

File: crates/validation/src/common/token_pricing.rs (hunk 1 above)
Comment (PlamenHristov) [reply crates/validation/src/common/token_pricing.rs:16]: Whenever we are working with the testnet setup you end up in a situation where you don't have enough blocks to fetch the price and on init the test nodes go boom 💥 !

File: testing-tools/src/controllers/devnet/config.rs (hunk 3 above)
Comment (PlamenHristov) [reply testing-tools/src/controllers/devnet/config.rs:89]: Sec

File: crates/validation/src/common/token_pricing.rs (hunk 1 above)
Comment (Will-Smith11) [reply crates/validation/src/common/token_pricing.rs:16]: ahh makes sense. Can we change it to where you pass a optional Override value (put override as the var name) in and then have it unwrap to the constant instead so it doesn't imply that its configurable
